eclipse-theia/theia
# fix #7100: remove `@theia/languages` and `monaco-languageclient`

#### What it does


- fix #7100: remove `@theia/languages` and `monaco-languageclient` - It allows to consume Monaco directly without trying to run vscode-languageclient in the browser. One should contribute language smartness via VS Code extensions instead.
- It cannot be part of 1.4.0, at least requires 1.5.0

TODO: 
- [x] following services should be deprecated in 1.4.0, see https://github.com/eclipse-theia/theia/pull/8139:
  - Language -> use `Language` from `@theia/core/lib/browser/language-service`
  - Languages -> use `monaco.languages.register*`, `MonacoLanguages.registerWorkspaceSymbolProvider` or `LanguageService` from `@theia/core/lib/browser/language-service`
  - Workspace -> use `MonacoWorkspace`
  - semantic highlighting -> VS Code extension after https://github.com/eclipse-theia/theia/pull/8010 and https://github.com/eclipse-theia/theia/issues/7608
  - call hierarchy -> VS Code extension
  - type hierarchy -> `TypeHierarchyRegistry.register` for now, later VS Code extension: https://github.com/microsoft/vscode/issues/15533
  - `languageServer` namespace -> use vscode-langaugeserver via VS Code extension

#### How to test


- [x] workspace symbols provided by VS Code extensions should be available:
  - wait till ts is initialized after opening a file
  - trigger `Go to Workspace Symbol...` command
  - type some prefix, wait till symbols are found
  - select some symbol and navigate to it
- [x] suggestion in the debug console should be properly provided and applied
  - put a breakpoint in uri.spec.ts
  - debug with `Run Mocha Test`
  - use the debug console to lookup variables
  - use different prefixes, apply suggestions and check that they resolve to proper variable names
- [x] language status bar entry should work
  - open some file
  - try to change a language
  - try to auto detect
- [x] git blame hover should work
  - open some file
  - trigger blame annotations
  - hover over them and check that hover corresponds to commit in the ruler
- [x] markdown preview widget should be in sync with text editors as before
  - open several markdown files and preview them (make sure to use preview from Theia, not built-in extension, one with old icon)
  - check that scrolling between preview and editor in sync
  - check that dirty changes are reflected
  - check that closed dirty editor without save reverts dirty changes in the preview
- [x] call hierarchies contributed by VS Code extensions should work
  - open ts file and wait till ts server is ready
  - trigger `Open Call Hierarchy` on some method
- [x] problem markers should be properly reflected
  - open some ts file
  - break it, should see markers in the problems view
  - delete it, markers should be removed
  - revert it in the scm view, markers should be added back
  - revert the dirty changes, markers should be removed

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
